server/app/services/geo/esri/EsriServiceAreaValidationConfig.java

```diff
@@ -0,0 +1,124 @@
+package services.geo.esri;
+
+import com.google.common.collect.ImmutableList;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigList;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/** Provides methods for handling Esri address service area validation config. */
+public final class EsriServiceAreaValidationConfig {
+  public Optional<ConfigList> ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS;
+  public Optional<ConfigList> ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES;
+  public Optional<ConfigList> ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS;
+  public Optional<ConfigList> ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS;
+
+  @Inject
+  public EsriServiceAreaValidationConfig(Config configuration) {
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS =
+        configuration.hasPath("esri_address_service_area_validation_labels")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_labels"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES =
+        configuration.hasPath("esri_address_service_area_validation_values")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_values"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS =
+        configuration.hasPath("esri_address_service_area_validation_urls")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_urls"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS =
+        configuration.hasPath("esri_address_service_area_validation_paths")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_paths"))
+            : Optional.empty();
+  }
+
+  /** Checks if each element necessary for Esri address service area validation is present. */
+  public Boolean hasAllElements() {
+    if (this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS.isEmpty()
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES.isEmpty()
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS.isEmpty()
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS.isEmpty()) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private Optional<ImmutableList<String>> getConfigListSetting(Optional<ConfigList> setting) {
+    if (setting.isEmpty()) {
+      return Optional.empty();
+    }
```

Comment:
**suggestion (blocking):** Add error logging anywhere we're returning an empty optional because of an error case

Reply:
added logging